pkg/controller/authentication/authentication_controller.go

```diff
@@ -0,0 +1,269 @@
+	idpSecret, err := getIdpSecret(ctx, r.client, authentication)
+	if err != nil {
+		log.Error(err, "Invalid or missing idp secret")
+		r.status.SetDegraded("Invalid or missing idp secret", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Set namespace for secrets so they can be used in the namespace of dex.
+	idpSecret = render.CopySecrets(render.DexNamespace, idpSecret)[0]
+
+	dexSecret := &corev1.Secret{}
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.DexObjectName, Namespace: render.OperatorNamespace()}, dexSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// We need to render a new one.
+			dexSecret = render.CreateDexClientSecret()
+		} else {
+			log.Error(err, "Failed to read dex secret")
+			r.status.SetDegraded("Failed to read dex secret", err.Error())
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Extract the connector to be configured for Dex.
+	pullSecrets, err := utils.GetNetworkingPullSecrets(network, r.client)
+	if err != nil {
+		log.Error(err, "Error retrieving pull secrets")
+		r.status.SetDegraded("Error retrieving pull secrets", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// DexConfig adds convenience methods around dex related objects in k8s and can be used to configure Dex.
+	dexCfg := render.NewDexConfig(authentication, tlsSecret, dexSecret, idpSecret)
+
+	// Create a component handler to manage the rendered component.
+	hlr := utils.NewComponentHandler(log, r.client, r.scheme, authentication)
+
+	// Render the desired objects from the CRD and create or update them.
+	reqLogger.V(3).Info("rendering components")
+	component := render.Dex(
+		pullSecrets,
+		r.provider == oprv1.ProviderOpenShift,
+		network,
+		dexCfg,
+	)
+
+	if err := hlr.CreateOrUpdate(context.Background(), component, r.status); err != nil {
+		log.Error(err, "Error creating / updating resource")
+		r.status.SetDegraded("Error creating / updating resource", err.Error())
+		return reconcile.Result{}, err
+	}
+
+	// Clear the degraded bit if we've reached this far.
+	r.status.ClearDegraded()
+
+	if !r.status.IsAvailable() {
+		// Schedule a kick to check again in the near future.
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	// Everything is available - update the CRD status.
+	authentication.Status.State = oprv1.TigeraStatusReady
+	if err = r.client.Status().Update(ctx, authentication); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func getIdpSecret(ctx context.Context, client client.Client, authentication *oprv1.Authentication) (*corev1.Secret, error) {
+	var secretName string
+
+	if authentication.Spec.OIDC != nil {
+		secretName = render.OIDCSecretName
+	} else {
+		secretName = render.OpenshiftSecretName
+	}
+
+	secret := &corev1.Secret{}
+	if err := client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: render.OperatorNamespace()}, secret); err != nil {
+		return nil, fmt.Errorf("missing secret %s: %w", secretName, err)
```

Comment:
generally good to include namespace when identifying namespaced objects in logs / errors.